### PR TITLE
support multiple loss average/broadcast

### PR DESCRIPTION
**Pull request by Meiyim**
# 功能更新： 当网络返回多个loss时，对各个loss做相应的 average、broadcast 并打印。

假设模型的loss计算如下：
```python
tensor_c = tensor_a + tensor_b
```
当网络需要返回多个loss的时候，网络的输出如下：

```python
{
    "loss": {"loss1": tensor_a, "loss2", tensor_b, "loss3", "loss": tensor_c},
    "logits": xxx
}
```

```python
{
    "loss": "tensor_a",
    "loss2": "tensor_b",
    "logits": xxx
}
```
 
 
这个时候框架会优化`"loss"` 对应的tensor ，并对其余tensor，进行累计、打印。

这个功能有网络返回 dict 结构时生效，对其他情况没有影响。

> TODO: 目前evaluate时，只打印`"loss"`，其余字段会忽略，后续支持多loss。

### CODE REVIEW

File: paddlenlp/trainer/trainer.py [hunk 1]
```diff
@@ -1817,17 +1827,19 @@ def training_step(self, model: nn.Layer, inputs: Dict[str, Union[paddle.Tensor,
         with self.autocast_smart_context_manager():
             loss, outputs = self.compute_loss(model, inputs, return_outputs=True)
         if self.args.gradient_accumulation_steps > 1:
-            loss = loss / self.args.gradient_accumulation_steps
+            loss = map_structure(lambda x: x / self.args.gradient_accumulation_steps, loss)
```
Comment (ZHUI) [inline paddlenlp/trainer/trainer.py:1829]:
```python
class NestedLoss(paddle.Tensor):
    def __init__(self, tensors):
        self.tensors=tensors

    def __div__(self, *args, **kwargs):
        return map_structure(lambda x:  x.__div__(*args, **kwargs), self.tensors)

    def detach(self, *args, **kwargs):
        return map_structure(lambda x:  x.detach(*args, **kwargs), self.tensors)

    def __getattr__(self, name: str) -> Any:
        return map_structure(lambda x:  getattr(name, x)(*args, **kwargs), self.tensors)

```

Comment (Meiyim) [reply paddlenlp/trainer/trainer.py:1829]: 后续 fix

**Review by ZHUI — APPROVED** [review]
LGTM
